src/main/java/com/squareup/kotlinpoet/TypeSpec.kt

```diff
@@ -101,6 +102,10 @@ class TypeSpec private constructor(builder: TypeSpec.Builder) {
         codeWriter.emitModifiers(modifiers, setOf(PUBLIC))
         if (kind == Kind.ANNOTATION) {
           codeWriter.emit("annotation class %L", name)
+        } else if (kind == Kind.OBJECT) {
+          codeWriter.emit("object class %L", name)
+        } else if (kind == Kind.COMPANION) {
+          codeWriter.emit("companion object")
```

Comment:
good question, it's not a feature i've ever used, but it wouldn't be hard to include